src/Stratis.Bitcoin.Features.PoA/BasePoAFeatureConsensusRules/PoAHeaderSignatureRule.cs

```diff
@@ -72,14 +73,14 @@ public override async Task RunAsync(RuleContext context)
             }
 
             // Look at the last round of blocks to find the previous time that the miner mined.
-            uint roundTime = this.slotsManager.GetRoundLengthSeconds(federation.Count);
+            var roundTime = TimeSpan.FromSeconds(this.slotsManager.GetRoundLengthSeconds(federation.Count));
```

Comment:
Doesn't `GetRoundLengthSeconds` already return seconds? Or are you doing `TimeSpan` cause of the uint return?

Reply:
TimeSpan is more granular than seconds.

Reply:
Why dont we then just refactor `GetRoundLengthSeconds` to return `TimeSpan`?

Reply:
Done.